### ISSUE TITLE
AttributeError: 'NoneType' object has no attribute 'close'

### ISSUE THREAD

**Issue by cthart** [nascheme/scgi#2]
Not sure if this is an issue with scgi or quixote but we're getting the following reproducible error when trying to serve two concurrent requests.
Python 3.8.10 on Ubuntu 20.04.3 LTS, all updates installed
scgi 2.1
Quixote 3.5:

```
Traceback (most recent call last):
  File "./server.py", line 260, in <module>
    run(run_l, port = config.webport, script_name='')
  File "/home/colin/venv/lib/python3.8/site-packages/quixote/server/scgi_server.py", line 55, in run
    s.serve()
  File "/home/colin/venv/lib/python3.8/site-packages/scgi/scgi_server.py", line 223, in serve
    self.serve_on_socket(sock)
  File "/home/colin/venv/lib/python3.8/site-packages/scgi/scgi_server.py", line 212, in serve_on_socket
    self.delegate_request(conn)
  File "/home/colin/venv/lib/python3.8/site-packages/scgi/scgi_server.py", line 189, in delegate_request
    self.reap_children()
  File "/home/colin/venv/lib/python3.8/site-packages/scgi/scgi_server.py", line 134, in reap_children
    child.close()
AttributeError: 'NoneType' object has no attribute 'close'
```

Reproduce by running
`for i in {1..1000}; do curl -S -s -o /dev/null http://server.domain.com/ ; done`
and doing another concurrent request to the same server.

I can't reproduce this on Python 2, or any of the other apps we have written using Quixote.

**Comment by cthart** [comment]
This patch fixes the problem for me:

```
--- /home/colin/venv/lib/python3.8/site-packages/scgi/scgi_server.py.orig	2021-09-07 17:15:03.635396407 +0000
+++ /home/colin/venv/lib/python3.8/site-packages/scgi/scgi_server.py.patched	2021-09-07 17:14:44.223138972 +0000
@@ -131,8 +131,9 @@
             if pid <= 0:
                 break
             child = self.get_child(pid)
-            child.close()
-            self.children.remove(child)
+            if child:
+                child.close()
+                self.children.remove(child)
 
     def do_stop(self):
         # Close connections to the children, which will cause them to exit
```

**Comment by nascheme** [comment]
Should be fixed in 2.2.
